enterprise/backup/src/main/java/org/neo4j/backup/BackupCommandArgumentHandler.java

```diff
@@ -0,0 +1,172 @@
+    private static final String ARG_DESC_CHECK_CONSISTENCY = "If a consistency check should be made.";
+
+    private static final String ARG_NAME_CHECK_GRAPH = "cc-graph";
+    private static final String ARG_DESC_CHECK_GRAPH = "Perform consistency checks between nodes, relationships, properties, types and tokens.";
+
+    private static final String ARG_NAME_CHECK_INDEXES = "cc-indexes";
+    private static final String ARG_DESC_CHECK_INDEXES = "Perform consistency checks on indexes.";
+
+    private static final String ARG_NAME_CHECK_LABELS = "cc-label-scan-store";
+    private static final String ARG_DESC_CHECK_LABELS = "Perform consistency checks on the label scan store.";
+
+    private static final String ARG_NAME_CHECK_OWNERS = "cc-property-owners";
+    private static final String ARG_DESC_CHECK_OWNERS = "Perform additional consistency checks on property ownership. This check is *very* expensive in time and memory.";
+
+    private static final Arguments arguments = new Arguments()
+        .withArgument( new MandatoryCanonicalPath( ARG_NAME_BACKUP_DIRECTORY, "backup-path", ARG_DESC_BACKUP_DIRECTORY ) )
+        .withArgument( new MandatoryNamedArg( ARG_NAME_BACKUP_NAME, "graph.db-backup", ARG_DESC_BACKUP_NAME ) )
+        .withArgument( new OptionalNamedArg( ARG_NAME_BACKUP_SOURCE, "address", "localhost:6362", ARG_DESC_BACKUP_SOURCE ) )
+        .withArgument( new OptionalBooleanArg( ARG_NAME_FALLBACK_FULL, true, ARG_DESC_FALLBACK_FULL ) )
+        .withArgument( new OptionalNamedArg( ARG_NAME_TIMEOUT, "timeout", "20m", ARG_DESC_TIMEOUT ) )
+        .withArgument( new OptionalBooleanArg( ARG_NAME_CHECK_CONSISTENCY, true, ARG_DESC_CHECK_CONSISTENCY ) )
+        .withArgument( new OptionalCanonicalPath( ARG_NAME_REPORT_DIRECTORY, "directory", ".", ARG_DESC_REPORT_DIRECTORY ) )
+        .withArgument( new OptionalCanonicalPath( ARG_NAME_ADDITIONAL_CONFIG_DIRECTORY, "config-file-path", "", ARG_DESC_ADDITIONAL_CONFIG_DIRECTORY ) )
+        .withArgument( new OptionalBooleanArg( ARG_NAME_CHECK_GRAPH, true, ARG_DESC_CHECK_GRAPH ) )
+        .withArgument( new OptionalBooleanArg( ARG_NAME_CHECK_INDEXES, true, ARG_DESC_CHECK_INDEXES ) )
+        .withArgument( new OptionalBooleanArg( ARG_NAME_CHECK_LABELS, true, ARG_DESC_CHECK_LABELS ) )
+        .withArgument( new OptionalBooleanArg( ARG_NAME_CHECK_OWNERS, false, ARG_DESC_CHECK_OWNERS ) );
+
+    BackupCommandArgumentHandler()
+    {
+    }
+
+    public static Arguments arguments()
+    {
+        return arguments;
+    }
+
+    CheckConsistencyConfig establishConfigurationOverride( Config config ) throws IncorrectUsage
+    {
+        boolean checkGraph;
+        boolean checkIndexes;
+        boolean checkLabelScanStore;
+        boolean checkPropertyOwners;
+        try
+        {
+            // We can remove the loading from config file in 4.0
+            if ( arguments.has( "cc-graph" ) )
+            {
+                checkGraph = arguments.getBoolean( "cc-graph" );
```

Comment:
Many opportunities to use the new constants here.

Reply:
Addressed in next push